packages/web/src/services/audius-backend/Jupiter.ts

```diff
@@ -77,39 +58,95 @@ const getQuote = async ({
   }
   const amount =
     swapMode === 'ExactIn'
-      ? JSBI.BigInt(Math.ceil(inputAmount * 10 ** inputToken.decimals))
-      : JSBI.BigInt(Math.floor(inputAmount * 10 ** outputToken.decimals))
-  const jup = await getInstance()
-  const routes = await jup.computeRoutes({
-    inputMint: new PublicKey(inputToken.address),
-    outputMint: new PublicKey(outputToken.address),
+      ? Math.ceil(inputAmount * 10 ** inputToken.decimals)
+      : Math.floor(inputAmount * 10 ** outputToken.decimals)
+  const jup = getInstance()
+  const quote = await jup.quoteGet({
+    inputMint: inputToken.address,
+    outputMint: outputToken.address,
     amount,
-    slippage,
+    slippageBps: slippage, // make sure slippageBps = slippage
+    asLegacyTransaction: true,
     swapMode,
-    forceFetch,
     onlyDirectRoutes
   })
-  const bestRoute = routes.routesInfos[0]
 
-  const resultQuote = {
-    inputAmount: convertJSBIToAmountObject(
-      bestRoute.inAmount,
+  if (!quote) {
+    throw new Error('Failed to get Jupiter quote')
+  }
+
+  return {
+    inputAmount: convertQuoteStringToAmountObject(
+      quote.inAmount,
       inputToken.decimals
     ),
-    outputAmount: convertJSBIToAmountObject(
-      bestRoute.outAmount,
+    outputAmount: convertQuoteStringToAmountObject(
```

Comment:
Why not make `quote.inAmount` and `quote.outAmount` into bigints, and use `convertBigIntToAmountObject`? (or vice versa, feels weird to need both converters)

Reply:
to quote dylan - bro you're so right